python/ribasim/ribasim/input_base.py

```diff
@@ -216,8 +216,13 @@ def tablename(cls) -> str:
     @model_validator(mode="before")
     @classmethod
     def _check_dataframe(cls, value: Any) -> Any:
+        # Enable initialization with a Dict.
+        if isinstance(value, dict) and len(value) > 0 and "df" not in value:
+            value = DataFrame(dict(**value))
```

Comment:
What does this look like when used, and why was this done? Is it tested?

Reply:
This replaces the `__init__` boilerplate for each nodetable (in the nodes folder, old code below). It's tested everytime you do something like `basin.Time(time=[])`.

```python
class Time(TableModel[BasinTimeSchema]):
    def __init__(self, **kwargs):
        super().__init__(df=DataFrame(dict(**kwargs)))
```

Reply:
Oh right I see, that's much nicer to define that once.